internal/resource/http.go

```diff
@@ -212,6 +212,8 @@ func (f *Fetcher) RewriteCAsWithDataUrls(cas []types.Resource) error {
 
 		encoded := dataurl.EncodeBytes(blob)
 		cas[i].Source = &encoded
+		// the rewrite wipes the compression
+		cas[i].Compression = nil
```

Comment:
This is similar to clearing `HTTPHeaders`, so maybe they should be grouped together.